src/workerd/io/hibernation-manager.h

```diff
@@ -176,9 +182,27 @@ class HibernationManagerImpl final : public Worker::Actor::HibernationManager {
   uint16_t hibernationEventType;
   // Passed to HibernatableWebSocket custom event as the typeId.
 
-  kj::Maybe<HibernatableWebSocket&> webSocketForEventHandler;
-  // Allows the HibernatableWebSocket event handler that is currently running to access the
-  // HibernatableWebSocket that it needs to execute.
+  const uint16_t EVENT_QUEUE_CAPACITY = 1024;
+  // Assume that at most 1024 events will dispatch and add websockets to the queue before events
+  // actually get their websockets.
+  kj::Vector<uint16_t> availableEventIDs;
+  // A collection of uints we use as ephemeral IDs for HibernatableWebSockets.
+  // Each ID acts as a key in the `webSocketsForEventHandler` map.
+  //
+  // When an event is to be dispatched, we pop a value off the back of `availableEventIDs`
+  // and insert it into `webSocketsForEventHandler`. Once the websocket has been retrieved by the
+  // event, we remove it from `webSocketsForEventHandler` and push the ID back into
+  // `availableEventIDs`. Since we're push/popping from the back we don't need to shift elements.
+  // Note that we don't need to retain sort order, we only need to guarantee IDs are unique to a
+  // hibernatable websocket.
```

Comment:
I'm open to other idea's (UUID), didn't want to defer to some crypto library off rip (or use strings)

Reply:
Ought to be able to make use of `workerd/util/uuid.h` easily enough here.

Reply:
https://github.com/cloudflare/workerd/pull/675/commits/75e84873cd9a34e7bc52789be66959d81588ac19

There's a lot of string copying going on so I'm going to look into decreasing that; wanted to get something checked in first though.